src/SignalRServiceExtension/Config/ServiceHubContextStore.cs

```diff
@@ -1,38 +1,52 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
 using System.Collections.Concurrent;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Azure.SignalR;
 using Microsoft.Azure.SignalR.Management;
-using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
 {
     internal class ServiceHubContextStore : IInternalServiceHubContextStore
     {
         private readonly ConcurrentDictionary<string, (Lazy<Task<IServiceHubContext>> lazy, IServiceHubContext value)> store = new ConcurrentDictionary<string, (Lazy<Task<IServiceHubContext>>, IServiceHubContext value)>(StringComparer.OrdinalIgnoreCase);
-        private readonly ILoggerFactory loggerFactory;
-        private readonly IOptionsMonitor<ServiceManagerOptions> monitor;
+        private readonly IOptionsMonitor<ContextOptions> monitor;
+        private readonly IServiceProvider serviceProvider;
 
-        public IServiceManager ServiceManager { get; }
+        public IServiceContext ServiceContext { get; }
 
-        public AccessKey[] AccessKeys => new AccessKey[] { new ServiceEndpoint(monitor.CurrentValue.ConnectionString).AccessKey };
+        public AccessKey[] AccessKeys => monitor.CurrentValue.ServiceEndpoints.Select(endpoint => endpoint.AccessKey).ToArray();
 
-        public ServiceHubContextStore(IOptionsMonitor<ServiceManagerOptions> optionsMonitor, IServiceManager serviceManager, ILoggerFactory loggerFactory)
+        // only create it on demand
+        public IServiceManager ServiceManager
```

Comment:
how about obsolete it

Reply:
`ServiceManager` is obsolete in the `IServiceHubContextStore`.